linera-views/src/key_value_store_view.rs

```diff
@@ -146,20 +170,38 @@ where
         self.was_cleared = true;
         self.updates.clear();
         self.deleted_prefixes.clear();
+        self.total_size = (0, 0);
+        self.sizes.clear();
         *self.hash.get_mut() = None;
     }
 }
 
 impl<'a, C> KeyValueStoreView<C>
 where
-    C: Send + Context,
+    C: Send + Context + Sync,
     ViewError: From<C::Error>,
 {
     fn max_key_size(&self) -> usize {
         let prefix_len = self.context.base_key().len();
         C::MAX_KEY_SIZE - 1 - prefix_len
     }
 
+    /// Getting the total sizes that will be used for keys and values when stored
+    /// ```rust
+    /// # tokio_test::block_on(async {
+    /// # use linera_views::memory::create_memory_context;
+    /// # use linera_views::key_value_store_view::KeyValueStoreView;
+    /// # use crate::linera_views::views::View;
+    /// # let context = create_memory_context();
+    ///   let mut view = KeyValueStoreView::load(context).await.unwrap();
+    ///   let total_size = view.total_size();
+    ///   assert_eq!(total_size, (0,0));
+    /// # })
+    /// ```
+    pub fn total_size(&self) -> (u64, u64) {
```

Comment:
See other comment. If we don't create a struct, then I would suggest two functions.

Reply:
I much prefer the struct way.